autogpt_platform/backend/backend/server/conn_manager.py

```diff
@@ -92,6 +101,21 @@ async def send_execution_update(
 
         return n_sent
 
+    async def send_notification(
+        self, *, user_id: str, payload: dict[str, Any] | list[Any] | str
+    ) -> int:
+        """Send a notification to all websocket connections belonging to a user."""
+        message = WSMessage(
+            method=WSMethod.NOTIFICATION,
+            data=payload,
+        ).model_dump_json()
+
+        connections = self.user_connections.get(user_id, set())
```

Comment:
nit: better self.user_connections.get(user_id) or set() so you don't keep creating a set.